lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,458 @@
+	}
+}
+
+// WithAdServicePrincipalId Set Active Directory principal id
+func WithAdServicePrincipalId(AdServicePrincipalId string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.AdServicePrincipalId = AdServicePrincipalId
+	}
+}
+
+// WithManagementGroup Enable the Management Group to allow AD to be reader on management group
+// rather then subscription
+func WithManagementGroup(enableManagentGroup bool) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.ManagementGroup = enableManagentGroup
+	}
+}
+
+// WithManagementGroupId The Group Id to add reader permissions
+func WithManagementGroupId(managementGroupId string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.ManagementGroupId = managementGroupId
+	}
+}
+
+// WithSubscriptionIds List of subscriptions to to enable logging
+func WithSubscriptionIds(subscriptionIds []string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.SubscriptionIds = subscriptionIds
+	}
+}
+
+// WithAllSubscriptions Grant read access to ALL subscriptions within
+// the selected Tenant (overrides 'subscription_ids')
+func WithAllSubscriptions(allSubscriptions bool) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.AllSubscriptions = allSubscriptions
+	}
+}
+
+// WithExistingStorageAccount Use an existing Storage Account
+func WithExistingStorageAccount(existingStorageAccount bool) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.ExistingStorageAccount = existingStorageAccount
+	}
+}
+
+// WithStorageAccountName The name of the Storage Account
+func WithStorageAccountName(storageAccountName string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.StorageAccountName = storageAccountName
+	}
+}
+
+// WithStorageAccountResourceGroup The Resource Group for the existing Storage Account
+func WithStorageAccountResourceGroup(storageAccountResourceGroup string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.StorageAccountResourceGroup = storageAccountResourceGroup
+	}
+}
+
+// WithLocation The Azure region where storage account for logging is
+func WithLocation(location string) AzureTerraformModifier {
+	return func(c *GenerateAzureTfConfigurationArgs) {
+		c.Location = location
+	}
+}
+
+// Generate new Terraform code based on the supplied args.
+func (args *GenerateAzureTfConfigurationArgs) Generate() (string, error) {
+	// Validate inputs
+	if err := args.validate(); err != nil {
+		return "", errors.Wrap(err, "invalid inputs")
+	}
+
+	// Create blocks
+	requiredProviders, err := createRequiredProviders()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate required providers")
+	}
+
+	azureADProvider, err := createAzureADProvider()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate AD provider")
+	}
```

Comment:
Should we be calling this here and also creating the AzureAdProvider in `createRequiredProviders()`?

Reply:
Don't think that this matters really, I made a create function for each individual section that is detailed in the example terraform from the help files